Extended readme.md

orangeDK opened the pull request:
Added a new chapter about flashing an image for Charge Control C and Charge Control M products.

orangeDK committed:
Extended readme.md

Added a new chapter about flashing an image for Charge Control C and
Charge Control M products.

Signed-off-by: David Krawehl <david.krawehl@chargebyte.com>

barsnick commented on README.md:
```diff
@@ -14,6 +14,7 @@ For problems and inquiries: https://tickets.in-tech-smartcharging.com/servicedes
 3.2 [Setting up the Yocto build environment](#Setting)  
 3.3 [Adding or removing layers](#addorremove)  
 3.4 [Building an Image](#build)  
+3.5 [Flashing an image](#flash)  
```
"Image" (capital) in 3.4, vs. "image" (non-capital) in 3.5.

mooraby commented on README.md:
```diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
+
+### Flashing an image <a name="flash"></a>
+
+The internal storage of the products Charge Control C and M is divided into several partitions and uses rauc to handle updates. If you are interested in adding rauc to your own image you can take a look at raucs documentation https://rauc.readthedocs.io/en/latest/integration.html#yocto .To flash an image it is important to identify the currently not in use partition because this will be our targe ...
```
Would prefer to have RAUC in capital letters.
Also "RAUC's documentation"

mhei replied:
please use always uppercase "RAUC" since it is an abbreviation

mooraby commented on README.md:
````diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
+```
+4. Mount `/dev/mmcblk0p2` and replace `/sbin/init` binary file with the following commands. (Omitting this step will format logging and data partitions)
+```bash
+mount /dev/mmcblk0p2 /mnt
````
not sure if we need sudo for mount and umount on the target platform

orangeDK replied:
Not necessary.

mooraby commented on README.md:
````diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
+
+1. To Identify the currently booted partition execute the following command and look for the partition which is marked as `good`. For this example we will be on partition `/dev/mmcblk0p1` on a Charge Control C.
+```bash
+rauc status
````
It would be better if an example of `rauc status` output is added

mhei commented on README.md:
```diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
 3. Execute `bitbake core-image-minimal` to build the image.
 
-The resulting image will be found in `yocto/build/tmp/deploy/image/<machine>`.
+The resulting image will be found in `yocto/build/tmp/deploy/image/<machine>` have a filename similar to `core-image-minimal-platform-1234567890-rootfs.ext4.gz`
```
I suggest: ... will be found in ..., (comma here before) having a filename...

mhei commented on README.md:
```diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
-The resulting image will be found in `yocto/build/tmp/deploy/image/<machine>`.
+The resulting image will be found in `yocto/build/tmp/deploy/image/<machine>` have a filename similar to `core-image-minimal-platform-1234567890-rootfs.ext4.gz`
+
+### Flashing an image <a name="flash"></a>
```
here too: uppercase I in image

mhei commented on README.md:
```diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
+The internal storage of the products Charge Control C and M is divided into several partitions and uses rauc to handle updates. If you are interested in adding rauc to your own image you can take a look at raucs documentation https://rauc.readthedocs.io/en/latest/integration.html#yocto .To flash an image it is important to identify the currently not in use partition because this will be our targe ...
+
+**Charge Control M:**
+| **`Partition`** | **`Size`**  | Description |
```
formatting nitpicks: partition and size is both, marked as bold and code; description neither nor, I'd suggest to drop both

mhei commented on README.md:
```diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
+| `/dev/mmcblk0p7` | `128 MB` | Logging file system B (/var/log) |
+
+**Charge Control C:**
+| **`Partition`** | **`Size`**  | Description |
```
same as above

mhei commented on README.md:
```diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
+| `/dev/mmcblk0p6` | `128 MB` | Logging file system A (/var/log) |
+| `/dev/mmcblk0p7` | `128 MB` | Logging file system B (/var/log) |
+
+1. To Identify the currently booted partition execute the following command and look for the partition which is marked as `good`. For this example we will be on partition `/dev/mmcblk0p1` on a Charge Control C.
```
small i in identify

mhei commented on README.md:
````diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
+```bash
+mount /dev/mmcblk0p2 /mnt
+mv /mnt/sbin/init.orig /mnt/sbin/init
+umount /dev/mmcblk0p2 -l
````
I'd not recommend to use option -l here, the umount should work without - if not, then something is wrong

orangeDK committed:
Adjustments after review

Corrected capitalisation errors. Removed -l option from umount command.
Removed table formatting. Fixed punctuation error.

Signed-off-by: David Krawehl <david.krawehl@chargebyte.com>

orangeDK committed:
Adress review comments

Fixed minor orthography error. Added example output of 'rauc status'.

Signed-off-by: David Krawehl <david.krawehl@chargebyte.com>
